### PR TITLE
Verify that VariablePointerFeatures.* and SubgroupSizeControl.* features can be used

### PR DESCRIPTION
**Pull request by cmarcelo**

### CODE REVIEW

**Comment by cmarcelo** [comment]
Force pushed to fix the formatting issue pointed by CI-check-format.

File: samples/config_helper_vulkan.cc [hunk 1]
```diff
@@ -667,8 +667,7 @@ amber::Result ConfigHelperVulkan::CreateVulkanInstance(
     uint32_t engine_major,
     uint32_t engine_minor,
     std::vector<std::string> required_extensions,
-    bool disable_validation_layer,
-    bool show_version_info) {
```
Comment (dj2) [inline samples/config_helper_vulkan.cc:671]: Why was the `show_version_info` flag removed?

File: samples/config_helper_vulkan.cc [hunk 2]
```diff
@@ -699,11 +698,10 @@ amber::Result ConfigHelperVulkan::CreateVulkanInstance(
   }
 
   // If dumping driver info then add useful extensions, if available.
```
Comment (dj2) [inline samples/config_helper_vulkan.cc:700]: This comment doesn't seem relevant anymore?

File: samples/config_helper_vulkan.cc [hunk 3]
```diff
@@ -785,13 +789,24 @@ amber::Result ConfigHelperVulkan::CheckVulkanPhysicalDeviceRequirements(
...
+      if (feature == kSubgroupSizeControl &&
+          size_control.subgroupSizeControl == VK_TRUE) {
+        continue;
+      }
+      if (feature == kComputeFullSubgroups &&
+          size_control.computeFullSubgroups == VK_TRUE) {
         continue;
       }
+
+      return amber::Result("Device does not support all required features");
```
Comment (dj2) [inline samples/config_helper_vulkan.cc:809]: If the feature isn't one of the above, won't this fail incorrectly?

File: samples/config_helper_vulkan.cc (hunk 1 above)
Comment (cmarcelo) [reply samples/config_helper_vulkan.cc:671]: The flag was used in this function to decide whether or not require `VK_KHR_get_physical_device_properties2` extension. The first patch changes the function to always require that extension if available.

**Comment by cmarcelo** [comment]
@dj2 added a new patch (the first one) with a fix for a missing initialization that I think was leading to failures in CI. If you prefer I can move it to a PR of its own.

Could you trigger the CI bots to run again?

**Comment by dj2** [comment]
I think it's fine in here. Kicked off the bots, will wait to see if they're happy.

**Comment by cmarcelo** [comment]
Repushed with the formatting fix.

In the other failing CIs, I've noticed many reported failed tests have a WARNING: `../third_party/swiftshader/src/Vulkan/VkStringify.cpp:468 WARNING: Stringify(VkStructureType v) is out of date. Please update it to match vulkan/vulkan_core.h`, maybe that is tripping the tests?

**Comment by cmarcelo** [comment]
> Repushed with the formatting fix.
> 
> In the other failing CIs, I've noticed many reported failed tests have a WARNING: `../third_party/swiftshader/src/Vulkan/VkStringify.cpp:468 WARNING: Stringify(VkStructureType v) is out of date. Please update it to match vulkan/vulkan_core.h`, maybe that is tripping the tests?

@dj2 that above turned out to be the case, the missing VkStringify entry in SwiftShader. I wrote a patch for that in https://swiftshader-review.googlesource.com/c/SwiftShader/+/45428 and if/once that lands we can update Amber's DEPS file.

**Comment by cmarcelo** [comment]
@dj2 my changes landed in SwiftShader and it seems https://github.com/google/amber/pull/869 will include them. Once that lands, I can rebase this PR and if I tested things write, we should get rid of the failures due to the VkStringify warning.

**Comment by dj2** [comment]
Awesome, thank you very much for the patch and all the hard work to get the various pieces in line.

**Pull request merged**